dynet/devices.h

```diff
@@ -1,19 +1,16 @@
 #ifndef DYNET_DEVICES_H
 #define DYNET_DEVICES_H
 
+#define EIGEN_USE_THREADS
 #include <string>
+#include <unsupported/Eigen/CXX11/Tensor>
+
 #include "dynet/aligned-mem-pool.h"
 #include "dynet/cuda.h"
 
-namespace Eigen {
-  struct DefaultDevice;
-  class CudaStreamDevice;
-  struct GpuDevice;
-}
-
 namespace dynet {
 
-enum class DeviceType {CPU, GPU};
+enum class DeviceType {CPU, GPU, ThreadPool};
```

Comment:
A little confused here, it seems that ThreadPool is a sub-option for both CPU and GPU(correct me if I am wrong). But here it seems it is a third option after CPU and GPU. What's the relationship between it with CPU and GPU? I don't suggest to add an enum value here for ThreadPool.